include/deal.II/grid/tria_accessor.h

```diff
@@ -138,54 +147,65 @@ namespace TriaAccessorExceptions
    *
    * @ingroup Exceptions
    */
-  DeclException0 (ExcCellNotActive);
+  DeclExceptionMsg (ExcCellNotActive,
+                    "The operation you are attempting can only be performed for "
+                    "(cell, face, or edge) iterators that point to 'active' "
+                    "objects. 'Active' objects are those that do not have "
+                    "children (in the case of cells), or that are part of "
+                    "an active cell (in the case of faces or edges). However, "
+                    "the object on which you are trying the current "
+                    "operation is not 'active' in this sense.");
   /**
    * Trying to access the children of a cell which is in fact active.
    *
    * @ingroup Exceptions
    */
-  DeclException0 (ExcCellHasNoChildren);
+  DeclExceptionMsg (ExcCellHasNoChildren,
+                    "The operation you are attempting can only be performed for "
+                    "(cell, face, or edge) iterators that have children, "
+                    "but the object on which you are trying the current "
+                    "operation does not have any.");
   /**
    * Trying to access the parent of a cell which is in the coarsest level of
    * the triangulation.
    *
    * @ingroup Exceptions
    */
-  DeclException0 (ExcCellHasNoParent);
-//TODO: Write documentation!
-  /**
-   * @ingroup Exceptions
-   */
-  DeclException0 (ExcUnusedCellAsChild);
-//TODO: Write documentation!
+  DeclExceptionMsg (ExcCellHasNoParent,
+                    "The operation you are attempting can only be performed for "
+                    "(cell, face, or edge) iterators that have a parent object, "
+                    "but the object on which you are trying the current "
+                    "operation does not have one -- i.e., it is on the "
+                    "coarsest level of the triangulation.");
   /**
    * @ingroup Exceptions
    */
   DeclException1 (ExcCantSetChildren,
                   int,
-                  << "You can only set the child index if the cell has no "
-                  << "children, or clear it. The given "
+                  << "You can only set the child index if the cell does not "
+                  << "currently have children registered; or you can clear it. "
+                  << "the childn index. The given "
```

Comment:
What happened here? Do you want to remove the three words?

Reply:
Ah, bad things clearly happened here. I've pushed a fixed version.